crypto/ec/ecp_nistp384.c

```diff
@@ -0,0 +1,1937 @@
+
+    output[4] += output[8] >> 32;
+    output[3] += (output[8] & 0xffffffff) << 24;
+    output[3] += output[8] >> 8;
+    output[2] += (output[8] & 0xff) << 48;
+    output[2] -= output[8] >> 16;
+    output[1] -= ((output[8] & 0xffff) << 40);
+    output[2] += output[8] >> 48;
+    output[1] += (output[8] & 0xffffffffffff) << 8;
+
+    output[3] += output[7] >> 32;
+    output[2] += (output[7] & 0xffffffff) << 24;
+    output[2] += output[7] >> 8;
+    output[1] += (output[7] & 0xff) << 48;
+    output[1] -= output[7] >> 16;
+    output[0] -= ((output[7] & 0xffff) << 40);
+    output[1] += output[7] >> 48;
+    output[0] += (output[7] & 0xffffffffffff) << 8;
+
+    /*-
+     * output[k] < in[k] + 2^124 + 2^121
+     *           < in[k] + 2^125, for k < 6
+     * output[6] < in[6] + 2^120 + 2^100 + 2^96 + 2^57
+     *           < in[6] + 2^121
+     */
+
+    /* Carry 4 -> 5 -> 6 */
+    output[5] += output[4] >> 56;
+    output[4] &= 0x00ffffffffffffff;
+
+    output[6] += output[5] >> 56;
+    output[5] &= 0x00ffffffffffffff;
+
+    /*-
+     * output[6] < in[6] + 2^121 + 2^72 + 2^16
+     *           < in[6] + 2^122
+     *           < 2^128
+     */
+
+    /* Eliminate high parts of output[6] */
+    temp = output[6] >> 48;
+    output[6] &= 0x0000ffffffffffff;
+    
+    /* temp < 2^80 */
+
+    output[3] += temp >> 40;
+    output[2] += (temp & 0xffffffffff) << 16;
+    output[2] += temp >> 16;
+    output[1] += (temp & 0xffff) << 40;
+    output[1] -= temp >> 24;
+    output[0] -= (temp & 0xffffff) << 32;
+    output[0] += temp;
+
+    /*-
+     * output[k] < output_old[k] + 2^64 + 2^56
+     *           < in[k] + 2^124 + 2^121 + 2^64 + 2^56 , k < 4
+     */
+
+    /* Carry 0 -> 1 -> 2 -> 3 -> 4 -> 5 -> 6 */
+    output[1] += output[0] >> 56;   /* output[1] < output_old[1] + 2^72 */
+    output[0] &= 0x00ffffffffffffff;
+
+    output[2] += output[1] >> 56;   /* output[2] < output_old[2] + 2^72 + 2^16 */
+    output[1] &= 0x00ffffffffffffff;
+
+    output[3] += output[2] >> 56;   /* output[3] < output_old[3] + 2^72 + 2^16 */
+    output[2] &= 0x00ffffffffffffff;
+
+    /*-
+     * output[k] < output_old[k] + 2^72 + 2^16
+     *           < in[k] + 2^124 + 2^121 + 2^72 + 2^64 + 2^56 + 2^16
+     *           < in[k] + 2^125
+     *           < 2^128 , k < 4
+     */
+
+    output[4] += output[3] >> 56;   /*-
+                                     * output[4] < output_old[4] + 2^72 + 2^16
+                                     *           < 2^72 + 2^56 + 2^16
+                                     */
+    output[3] &= 0x00ffffffffffffff;
+
+    output[5] += output[4] >> 56;   /*-
+                                     * output[5] < output_old[5] + 2^16 + 1
+                                     *           < 2^56 + 2^16 + 1
+                                     */
+    output[4] &= 0x00ffffffffffffff;
+
+    output[6] += output[5] >> 56;   /* output[6] < 2^48 + 1 */
+    output[5] &= 0x00ffffffffffffff;
+
+    for (i = 0; i < NLIMBS; i++)
```

Comment:
`memcpy(3)` ?

Reply:
Members in `output` and `out` have different sizes, so I'll keep the loop of assignments here.